src/containers/Stake.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from "react";
+import {connect} from 'react-redux';
+import StakeForm from '../components/StakeForm';
+import {addContractData } from '../redux/actions/addReward';
+
+class Stake extends Component {
+    
+    state = {
+        amount: 0,
+    };
+
+    async componentDidMount(){
+        
+        const {addContractDataFunc, contract: {data}} = this.props;
+        if (!data){
+            await addContractDataFunc()
+        } 
+    };
+
+    allow=async(addr, amount)=> {
+        const {contract: {data:{owner, GAS, contractAddress, frm}}} = this.props;
+        console.log('Transferring');
+        await frm.methods.transfer(addr, amount).send({from: owner, gas: GAS});
+        console.log('Approving');
+        await frm.methods.approve(contractAddress, amount).send({from: addr});
```

Comment:
Use contract proxy

Make sure transaction is confirmed before running the next method